app_feup/test/integration/src/exams_page_test.dart

```diff
@@ -78,5 +83,60 @@ void main() {
       expect(find.byKey(Key(sdisExam.toString())), findsOneWidget);
       expect(find.byKey(Key(sopeExam.toString())), findsOneWidget);
     });
+
+    testWidgets('Filtered Exams', (WidgetTester tester) async {
+      final store = Store<AppState>(appReducers,
+          initialState: AppState({
+            'session': Session(authenticated: true),
+            'currUcs': [sopeCourseUnit, sdisCourseUnit],
+            'exams': List<Exam>(),
+            'profile': profile,
+            'filteredExams': filteredExams
+          }),
+          middleware: [generalMiddleware]);
+
+      NetworkRouter.httpClient = mockClient;
+
+      final mockHtml =
+          File('integration/resources/exam_example.html').readAsStringSync();
+      when(mockResponse.body).thenReturn(mockHtml);
+      when(mockResponse.statusCode).thenReturn(200);
+      when(mockClient.get(any, headers: anyNamed('headers')))
+          .thenAnswer((_) async => mockResponse);
+
+      final Completer<Null> completer = Completer();
+      final actionCreator =
+          getUserExams(completer, ParserExams(), Tuple2('', ''));
+
+      final newfilteredExams = Map<String, bool>.from(filteredExams);
+      newfilteredExams['Mini-testes'] = false;
+      final Completer<Null> completerFilter = Completer();
+
+      final actionCreatorFilter =
+          setFilteredExams(newfilteredExams, completerFilter);
+
+      final widget = testableReduxWidget(child: ExamsPageView(), store: store);
+
+      await tester.pumpWidget(widget);
+
+      expect(find.byKey(Key(sdisExam.toString())), findsNothing);
+      expect(find.byKey(Key(sopeExam.toString())), findsNothing);
+
+      actionCreator(store);
+
+      await completer.future;
+
+      await tester.pumpAndSettle();
+      expect(find.byKey(Key(sdisExam.toString())), findsOneWidget);
+      expect(find.byKey(Key(sopeExam.toString())), findsOneWidget);
+
+      actionCreatorFilter(store);
+
+      await completerFilter.future;
+
+      await tester.pumpAndSettle();
+      expect(find.byKey(Key(sdisExam.toString())), findsNothing);
+      expect(find.byKey(Key(sopeExam.toString())), findsNothing);
```

Comment:
I think this integration test should be even more complicated. After what it is doing now, it should open the modal, change the types of exams it wants to see, and then make sure a new exam of that type appears. Wdyt?